### PR TITLE
Use Cgo callback because NewCallback would eventually SIGSEGV on amd64

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
This is a temporary fix and should be replaced by `NewCallback` once it is fixed. 

Updates #195

### CODE REVIEW

**Comment by hajimehoshi** [comment]
Thanks for your quick work!

Please check the test failures.

**Comment by hajimehoshi** [comment]
It's OK to merge this, but I can wait for your firx of purego.

**Comment by TotallyGamerJet** [comment]
I would merge this now so that the latest commit works properly.

File: api_darwin.go [hunk 1]
```diff
@@ -71,16 +71,19 @@ type _AudioQueueBuffer struct {
 
 type _AudioQueueOutputCallback func(inUserData unsafe.Pointer, inAQ _AudioQueueRef, inBuffer _AudioQueueBufferRef)
 
+var _AudioQueueNewOutput uintptr
+
 func init() {
 	toolbox := purego.Dlopen("/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox", purego.RTLD_GLOBAL)
-	purego.RegisterLibFunc(&_AudioQueueNewOutput, toolbox, "AudioQueueNewOutput")
+	//purego.RegisterLibFunc(&_AudioQueueNewOutput, toolbox, "AudioQueueNewOutput")
```
Comment (hajimehoshi) [inline api_darwin.go:78]: Let's remove the comment.

File: api_darwin.go [hunk 2]
```diff
@@ -71,16 +71,19 @@ type _AudioQueueBuffer struct {
...
+	_AudioQueueNewOutput = purego.Dlsym(toolbox, "AudioQueueNewOutput")
 	purego.RegisterLibFunc(&_AudioQueueAllocateBuffer, toolbox, "AudioQueueAllocateBuffer")
 	purego.RegisterLibFunc(&_AudioQueueEnqueueBuffer, toolbox, "AudioQueueEnqueueBuffer")
 	purego.RegisterLibFunc(&_AudioQueueStart, toolbox, "AudioQueueStart")
 	purego.RegisterLibFunc(&_AudioQueuePause, toolbox, "AudioQueuePause")
 }
 
-var _AudioQueueNewOutput func(inFormat *_AudioStreamBasicDescription, inCallbackProc _AudioQueueOutputCallback, inUserData unsafe.Pointer, inCallbackRunLoop uintptr, inCallbackRunLoopMod uintptr, inFlags uint32, outAQ *_AudioQueueRef) uintptr
+//var _AudioQueueNewOutput func(inFormat *_AudioStreamBasicDescription, inCallbackProc _AudioQueueOutputCallback, inUserData unsafe.Pointer, inCallbackRunLoop uintptr, inCallbackRunLoopMod uintptr, inFlags uint32, outAQ *_AudioQueueRef) uintptr
```
Comment (hajimehoshi) [inline api_darwin.go:86]: ditto

File: driver_darwin.go [hunk 3]
```diff
@@ -14,12 +14,20 @@
 
 package oto
 
+// #cgo LDFLAGS: -framework AudioToolbox
+//
+// #import <AudioToolbox/AudioToolbox.h>
+//
+// void oto_render(void* inUserData, AudioQueueRef inAQ, AudioQueueBufferRef inBuffer);
+//
```
Comment (hajimehoshi) [inline driver_darwin.go:22]: This is just my preference, but let's remove the last `//`.

**Comment by hajimehoshi** [comment]
> I would merge this now so that the latest commit works properly.

Do we have an estination how long it would take to fix the root issue?

**Comment by TotallyGamerJet** [comment]
Not really since were talking about analyzing assembly. I'm also going to be busy and only be able to work on it during my free time.

**Comment by hajimehoshi** [comment]
Sure, thanks!

**Review by hajimehoshi — APPROVED** [review]
LGTM

**Pull request merged**